### DISCUSSION TITLE
Browser not supported error on login

### DISCUSSION THREAD

**Discussion by D10ny5u5** [Zerg00s/FlowPowerAppsMigrator#37] [Q&A]
My company enforces Edge or Chrome for Office or authentication. I don't think they'd be willing to change this due to security concerns. Is there any way around this? Any chance of changing the code to "mimic" one of them?

I believe this is a MCAS issue (Renamed Microsoft Cloud App Security. It's now called Microsoft Defender for Cloud Apps)

![image](https://user-images.githubusercontent.com/15118504/203692733-2f145174-ad1f-4252-92c8-a832e0d551dd.png)

**Comment by Zerg00s** [comment]
Hi @D10ny5u5 this looks like a problem that might be commonplace soon. 

There are few potential workarounds. But I am not sure if it's going to work in your case. These workarounds require high permissions on the tenant level. What permission levels do you have at your Microsoft 365 tenant? Do you have Global Admin or SharePoint Admin role?

- Option 1: [Create a SharePoint-only app](https://www.youtube.com/watch?v=Tn-jB7vdnOE). Update the `Connect-PnPOnline` lines to something like this:
```
$appId = "0a20932b-bd9d-XXXX-8c5b-2ef9fc7ee72c"
$appSecret = "+xKIALIQSvnxNnAOu7yerkFxOZaVxXXXXKZZMUMX1X0="
$url = "https://contoso.sharepoint.com/"
Connect-PnPOnline -Url $url -ClientId $appId -ClientSecret $appSecret
```
- Option 2: Create Azure App Registration. Update the `Connect-PnPOnline` lines to something like this:
`Connect-PnPOnline 
    -Thumbprint '6CA93F4CA9C32A490361986AB3170EC8E1FAFFB9' `
    -Tenant contoso.onmicrosoft.com `
    -ClientId fb32c27b-57be-4824-8af0-e8d303553d9e `
    -Url https://contoso.sharepoint.com/
`
- Option 3. Update the `Connect-PnPOnline` lines to 
```
Connect-PnPOnline -Url "contoso.sharepoint.com" -DeviceLogin -LaunchBrowser
```
